charon/src/bin/charon-driver/translate/translate_types.rs

```diff
@@ -858,20 +856,39 @@ impl ItemTransCtx<'_, '_> {
                             unreachable!()
                         }
                     }
+                    // If there is exactly one variant, i.e. if the type is a struct,
+                    // there is no discriminant, but still fields with offsets.
+                    r_abi::Variants::Single { index } if *index == r_abi::VariantIdx::ZERO => {
```

Comment:
Can you turn this `if` into an `assert!`? I don't know of a case where this isn't true, and if there is one we should handle it properly.